uk.co.compucorp.civicrm.hrcore/tests/phpunit/CRM/HRCore/Form/Search/StaffDirectoryTest.php

```diff
@@ -0,0 +1,214 @@
+    $searchDirectory =  new SearchDirectory($formValues);
+
+    //Contact1 has future contract
+    $this->assertEquals(1, $searchDirectory->count());
+
+    //verify contact ids
+    $contactIds = $this->extractContactIds($searchDirectory->contactIDs()) ;
+    $this->assertEquals($contactIds, [$contact1['id']]);
+  }
+
+  public function testCountReturnsTheCorrectNumberOfStaffWithSpecificJobContractDates() {
+    $contact1 = ContactFabricator::fabricate();
+    $contact2 = ContactFabricator::fabricate();
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $contact1['id']],
+      [
+        'period_start_date' => '2016-01-01',
+        'period_end_date' => '2016-12-31'
+      ]
+    );
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $contact2['id']],
+      ['period_start_date' => '2018-01-01']
+    );
+
+    $formValues = [
+      'select_staff' => 'choose_date',
+      'contract_start_date' => '2016-04-01',
+      'contract_end_date' => '2016-05-01'
+    ];
+    $searchDirectory =  new SearchDirectory($formValues);
+
+    //only Contact1 has contract dates overlapping selected dates
+    $this->assertEquals(1, $searchDirectory->count());
+
+    //verify contact ids
+    $contactIds = $this->extractContactIds($searchDirectory->contactIDs()) ;
+    $this->assertEquals($contactIds, [$contact1['id']]);
+  }
+
+  public function testCountReturnsZeroWhenNoStaffWithContractsOverlappingSpecificJobContractDates() {
+    $contact1 = ContactFabricator::fabricate();
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $contact1['id']],
+      [
+        'period_start_date' => '2016-05-02',
+        'period_end_date' => '2016-12-31'
+      ]
+    );
+
+    $formValues = [
+      'select_staff' => 'choose_date',
+      'contract_start_date' => '2016-04-01',
+      'contract_end_date' => '2016-05-01'
+    ];
+    $searchDirectory =  new SearchDirectory($formValues);
+
+    //No staff with contract dates overlapping the contract dates selected
+    $this->assertEquals(0, $searchDirectory->count());
+  }
+
+  public function testCountReturnsTheCorrectNumberOfStaffWithRelativeJobContractDate() {
+    $contact1 = ContactFabricator::fabricate();
+    $contact2 = ContactFabricator::fabricate();
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $contact1['id']],
+      [
+        'period_start_date' => '2016-01-01',
+        'period_end_date' => '2016-12-31'
+      ]
+    );
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $contact2['id']],
+      ['period_start_date' => '2018-01-01']
+    );
+
+    $formValues = ['select_staff' => 'this.day'];
+    $searchDirectory =  new SearchDirectory($formValues);
+
+    //only Contact2 has contract dates overlapping today
+    $this->assertEquals(1, $searchDirectory->count());
+
+    //verify contact ids
+    $contactIds = $this->extractContactIds($searchDirectory->contactIDs()) ;
+    $this->assertEquals($contactIds, [$contact2['id']]);
+  }
```

Comment:
Seeing these tests, it looks more like you're testing the behavior of the search when using the "Select Staff" filter than testing the `count()` method, as the test names describe. So I suggest renaming the tests to reflect that.

Also, it would be good to have some tests for showing the returned fields are correct

Reply:
@tunbola the test names are still like "testCountReturns...", so they're still about count.My suggestion to rename them was to make it clear that it's all about the filter, so I would expect things more like `testSelectStaffFilterCanFilterOnlyContactsCurrentStaff`, `testSelectStaffFilterCanFilterOnFutureStaff`, etc.